testworld/document_consensus_test.go

```diff
@@ -2,76 +2,89 @@
+
+	// Bob updates invoice and shares with Charlie as well
+	res, err = bob.host.updateInvoice(bob.expect, http.StatusOK, docIdentifier, updatedInvoicePayload([]string{alice.id.String(), charlie.id.String()}))
 
-	// Alice shares invoice document with Bob first
-	res, err := alice.createInvoice(eAlice, map[string]interface{}{
-		"data": map[string]interface{}{
-			"invoice_number": "12324",
-			"due_date":       "2018-09-26T23:12:37.902198664Z",
-			"gross_amount":   "40",
-			"currency":       "GBP",
-			"net_amount":     "40",
-		},
-		"collaborators": []string{b.String()},
-	})
 	if err != nil {
 		t.Error(err)
 	}
-	docIdentifier := res.Value("header").Path("$.document_id").String().NotEmpty().Raw()
+	docIdentifier = getDocumentIdentifier(t, res)
+
 	if docIdentifier == "" {
 		t.Error("docIdentifier empty")
 	}
+	params["currency"] = "EUR"
+	getInvoiceAndCheck(alice.expect, params)
+	getInvoiceAndCheck(bob.expect, params)
+	getInvoiceAndCheck(charlie.expect, params)
+}
+
+func TestHost_CollaboratorTimeOut(t *testing.T) {
+
+	alice := getHostTestSuite(t, "Alice")
+	bob := getHostTestSuite(t, "Bob")
+
+	// alice shares an invoice bob
+	response, err := alice.host.createInvoice(alice.expect, http.StatusOK, defaultInvoicePayload([]string{bob.id.String()}))
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	// check if bob and alice received the document
+	docIdentifier := getDocumentIdentifier(t, response)
 	params := map[string]interface{}{
 		"document_id": docIdentifier,
-		"currency":    "GBP",
+		"currency":    "USD",
 	}
-	getInvoiceAndCheck(eAlice, params)
-	getInvoiceAndCheck(eBob, params)
+	getInvoiceAndCheck(alice.expect, params)
+	getInvoiceAndCheck(bob.expect, params)
 
-	// Bob updates invoice and shares with Charlie as well
-	res, err = bob.updateInvoice(eBob, docIdentifier, map[string]interface{}{
-		"data": map[string]interface{}{
-			"invoice_number": "12324",
-			"due_date":       "2018-09-26T23:12:37.902198664Z",
-			"gross_amount":   "40",
-			"currency":       "USD",
-			"net_amount":     "40",
-		},
-		"collaborators": []string{a.String(), c.String()},
-	})
+	// Alice gets killed
+	alice.host.canc()
+
+	// Bob updates and sends to Alice
+	updatedPayload := updatedInvoicePayload([]string{alice.id.String()})
+
+	// Bob will anchor the document without Alice signature but will receive an error because alice is dead
+	response, err = bob.host.updateInvoice(bob.expect, http.StatusInternalServerError, docIdentifier, updatedPayload)
 	if err != nil {
 		t.Error(err)
 	}
-	docIdentifier = res.Value("header").Path("$.document_id").String().NotEmpty().Raw()
-	if docIdentifier == "" {
-		t.Error("docIdentifier empty")
+
+	// check if bob saved the updated document
+	params = map[string]interface{}{
+		"document_id": docIdentifier,
+		"currency":    "EUR",
 	}
-	params["currency"] = "USD"
-	getInvoiceAndCheck(eAlice, params)
-	getInvoiceAndCheck(eBob, params)
-	getInvoiceAndCheck(eCharlie, params)
+	getInvoiceAndCheck(bob.expect, params)
```

Comment:
You need to restart any host that you bring down at the end of the test, otherwise other tests are affected.

Reply:
Also lets create a new host that is explicitly meant to be brought down for tests if required. Since I want to make sure that we can run all tests concurrently in the future.

Reply:
yes, you are right. I will add a own host for that